API_DESIGN.md

```diff
@@ -460,3 +460,49 @@ message VerifyEmailRequest{
 }
 
 ```
+
+## Deprecations
```

Comment:
Do you want to have a dedicated section or should we move it to / under `Versioning`